src/PowerShellEditorServices/Extensions/EditorFileRanges.cs

```diff
@@ -0,0 +1,371 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using Microsoft.PowerShell.EditorServices.Handlers;
+using Microsoft.PowerShell.EditorServices.Services.TextDocument;
+using OmniSharp.Extensions.LanguageServer.Protocol.Models;
+using System;
+
+namespace Microsoft.PowerShell.EditorServices.Extensions
```

Comment:
This file defiles a whole set of file APIs in both 1-based and 0-based indexing conventions and can convert between them all